### ISSUE TITLE
量化模型GPTQ推理速度慢

### ISSUE THREAD

**Issue by thuBingo** [QwenLM/Qwen2-VL#483]
使用 Qwen/Qwen2-VL-7B-Instruct-GPTQ-Int4 进行推理，代码如下，推理速度比普通的7B模型 Qwen/Qwen2-VL-7B-Instruct 慢一倍左右。推理实例图片分别耗时21秒和6秒。

使用GPTQ模型推理时，推理能够成功但会print: CUDA extension not installed.
You are attempting to use Flash Attention 2.0 without specifying a torch dtype. This might lead to unexpected behaviour
看测评的话，只慢一点点

from transformers import Qwen2VLForConditionalGeneration, AutoTokenizer, AutoProcessor
from qwen_vl_utils import process_vision_info
import torch

model = Qwen2VLForConditionalGeneration.from_pretrained(
    "Qwen/Qwen2-VL-7B-Instruct-GPTQ-Int4",
    torch_dtype="auto",
    attn_implementation="flash_attention_2",
    device_map={"": "cuda:1"},
)

processor = AutoProcessor.from_pretrained(
    "Qwen/Qwen2-VL-7B-Instruct-GPTQ-Int4"
)

messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "https://qianwen-res.oss-cn-beijing.aliyuncs.com/Qwen-VL/assets/demo.jpeg",
            },
            {"type": "text", "text": "Describe this image."},
        ],
    }
]

# Preparation for inference
import time
start_time = time.time()
text = processor.apply_chat_template(
    messages, tokenize=False, add_generation_prompt=True
)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[text],
    images=image_inputs,
    videos=video_inputs,
    padding=True,
    return_tensors="pt",
)
inputs = inputs.to("cuda:1")

# Inference: Generation of the output
generated_ids = model.generate(**inputs, max_new_tokens=128)
generated_ids_trimmed = [
    out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
]
output_text = processor.batch_decode(
    generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
)
print(output_text)
end_time = time.time()
print('inference time', end_time-start_time)

**Comment by thuBingo** [comment]
已经解决了，https://github.com/AutoGPTQ/AutoGPTQ/blob/main/README.md    根据 Install from source指南重新安装auto-gptq

**Comment by zzf2grx** [comment]
请问下， Qwen2-VL-7B-Instruct-GPTQ-Int4，相比Qwen2-VL-7B-Instruct有速度上的优势吗？

**Comment by gameboy777x** [comment]
> 请问下， Qwen2-VL-7B-Instruct-GPTQ-Int4，相比Qwen2-VL-7B-Instruct有速度上的优势吗？

Is there an answer now? Thank you very much
